datanode/bootstrap.go

```diff
@@ -121,56 +123,20 @@ func (m *bootstrapManagerImpl) Bootstrap() error {
 }
 
 func (m *bootstrapManagerImpl) bootstrap() error {
-	starDatanodeBootstrap := utils.Now()
-
-	tables := m.datanode.Tables()
-	shardSet := m.datanode.ShardSet()
-
-	workers := xsync.NewWorkerPool(m.opts.BootstrapOptions().MaxConcurrentTableShards())
-	workers.Init()
-
-	var (
-		multiErr = xerrors.NewMultiError()
-		mutex    sync.Mutex
-		wg       sync.WaitGroup
-	)
-
+	startDatanodeBootstrap := utils.Now()
 	topoStateSnapshot := newInitialTopologyState(m.topo)
-	for _, shard := range shardSet.All() {
-		shardID := shard.ID()
-		for _, table := range tables {
-			tableShard, err := m.datanode.GetTableShard(table, shardID)
-			if err != nil {
-				multiErr = multiErr.Add(err)
-				continue
-			}
-
-			if tableShard.IsBootstrapped() {
-				tableShard.Users.Done()
-				continue
-			}
-
-			wg.Add(1)
-			workers.Go(func() {
-				defer wg.Done()
-				err := tableShard.Bootstrap(m.peerSource, m.datanode.ID(), m.topo, topoStateSnapshot, m.opts.BootstrapOptions())
-				mutex.Lock()
-				multiErr = multiErr.Add(err)
-				mutex.Unlock()
-				// unpin table shard after use
-				tableShard.Users.Done()
-			})
-		}
-	}
-	wg.Wait()
-	if !multiErr.Empty() {
-		return multiErr.FinalError()
+	err := m.bootstrapable.Bootstrap(m.peerSource, m.origin, m.topo, topoStateSnapshot, m.opts.BootstrapOptions())
```

Comment:
is BootstrapManager and Bootstrap 1:1 ?
in that case BootstrapManager seems dummy and can be removed..

Reply:
bootstrapManager still acts as a scheduler